dubbo-common/src/main/java/org/apache/dubbo/common/beans/factory/ScopeBeanFactory.java

```diff
@@ -53,7 +54,9 @@ public class ScopeBeanFactory {
     private final List<BeanInfo> registeredBeanInfos = new CopyOnWriteArrayList<>();
     private InstantiationStrategy instantiationStrategy;
     private final AtomicBoolean destroyed = new AtomicBoolean();
-    private List<Class<?>> registeredClasses = new ArrayList<>();
+    private final List<Class<?>> registeredClasses = new ArrayList<>();
+
+    private final Map<Class<?>, Object> registeredCacheBeanMap = new ConcurrentHashMap<>(64);
```

Comment:
Cache here should consider different bean with same class but different name

Reply:
Only beans without names are cached here. If most beans have no names, is this enough? Maybe changing the name of the map to NoNameBeanMap is more appropriate

Reply:
Aactually is the only unique Name